### DISCUSSION TITLE
Independent lockscreen

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#784] [Enhancements]
Currently we rely on the platform's default locks creen to authenticate the user in case they've enabled the lock screen within app's settings or while performing sensitive actions.

We could give the user an option to choose an independent lockscreen (password / pin / pattern) to serve the use case where the device's authentication vector might be shared with someone else.

We could also leverage this interface in the future to set specific password / pin / pattern for each hidden folder.

**Comment by vishnukvmd** [comment]
Once implemented, we could detect changes to the biometric vectors and enforce our independent lockscreen to prevent unauthorized personnel who could have potentially tampered with the device from accessing the gallery.

Source: https://www.reddit.com/r/enteio/comments/14jhs4r/

**Comment by 6vers** [comment]
Maybe you could also add a option so you can only use biometrics and not the device code for unlocking the hidden folder in the photos app if a change in the biometrics would have been detected the code that was pre configured in the app would need to be used to enable biometrics again

**Comment by danielkrysiak** [comment]
This is how it's done in Tresorit. It allows setting a timeout, and automatic wipe on to many PIN attempts. 

I have a special use case, short version: I'm using separate user profile, which is completely unlocked. I'm taking pictures there, and I want to sync them with Ente. Ente doesn't allow me to set app locking without setting up system locking first. At this point, in practice, I'm forced to fully logout, then do complicated 2-step login process when I want to sync.

Long version:  I use a separate user profile on my Android phone for hiking (app installs blocked and cellular blocked - for those wondering). I hike in every weather conditions and I'm using this profile to navigate, listen to media, and take pictures. My gloves allow me to do only very simple tasks, but not to type a PIN, especially every time I want to check a map.

On a margin, the ultimate app for me would be as "sync only/one way" app, no giving access to the pictures at all.
